Content.Server/Physics/Controllers/ChasingWalkSystem.cs

```diff
@@ -0,0 +1,86 @@
+using System.Linq;
+using Content.Server.Physics.Components;
+using Robust.Shared.Random;
+using Robust.Shared.Timing;
+using Robust.Shared.Physics.Systems;
+using Robust.Shared.Physics.Components;
+
+namespace Content.Server.Physics.Controllers;
+
+/// <summary>
+/// A system which makes its entity chasing another entity with selected component.
+/// </summary>
+public sealed class ChasingWalkSystem : EntitySystem
+{
+    [Dependency] private readonly IGameTiming _gameTiming = default!;
+    [Dependency] private readonly IRobustRandom _random = default!;
+    [Dependency] private readonly EntityLookupSystem _lookup = default!;
+    [Dependency] private readonly SharedPhysicsSystem _physics = default!;
+
+    public override void Update(float frameTime)
+    {
+        base.Update(frameTime);
+
+        var query = EntityQueryEnumerator<ChasingWalkComponent>();
+        while (query.MoveNext(out var uid, out var chasing))
+        {
+            //Set Velocity to Target
+            if (chasing.NextImpulseTime <= _gameTiming.CurTime)
+            {
+                ForceImpulse(uid, chasing);
+                chasing.NextImpulseTime = _gameTiming.CurTime + TimeSpan.FromSeconds(chasing.ImpulseInterval);
+            }
+            //Change Target
+            if (chasing.NextChangeVectorTime <= _gameTiming.CurTime)
+            {
+                ChangeTarget(uid, chasing);
+
+                var delay = TimeSpan.FromSeconds(_random.NextFloat(chasing.ChangeVectorMinInterval, chasing.ChangeVectorMaxInterval));
+                chasing.NextChangeVectorTime = _gameTiming.CurTime + delay;
+            }
+        }
+    }
+
+    private void ChangeTarget(EntityUid uid, ChasingWalkComponent component)
+    {
+        //We find our coordinates and calculate the radius of the target search.
+        var xform = Transform(uid);
+        var range = component.MaxChaseRadius;
+        var compType = EntityManager.ComponentFactory.GetRegistration(component.ChasingComponent).Type;
+        var allEnts = _lookup.GetComponentsInRange(compType, xform.MapPosition, range)
+            .Select(x => x.Owner).ToList();
```

Comment:
Using nested LINQ is very expensive so someone should go over this PR and optimise it as a lot of it is more expensive than it needs to be.

Reply:
Also should be using compregistry instead / make a typeserializer rather than getting registration every time.

Reply:
need help with it
i fix first part, with LINQ .Select removing 
![image](https://github.com/space-wizards/space-station-14/assets/96445749/98cb2995-cb04-41b3-a519-10f0862bf4c1)

but second part with registraion sound confusing to me